packages/frontend/src/components/RefreshServer/RefreshStepsModal/index.tsx

```diff
@@ -0,0 +1,165 @@
+    const statusInfo = (
+        status: string,
+    ): { title: string; icon: IconName; status: string } => {
+        switch (status) {
+            case 'DONE':
+                return {
+                    title: 'Sync successful!',
+                    icon: 'tick-circle',
+                    status: 'Success',
+                };
+            case 'ERROR':
+                return {
+                    title: 'Error in sync',
+                    icon: 'warning-sign',
+                    status: 'Error',
+                };
+            case 'RUNNING':
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'In progress',
+                };
+
+            case 'PENDING':
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'Queued',
+                };
+
+            default:
+                return {
+                    title: 'Sync in progress',
+                    icon: 'refresh',
+                    status: 'Success',
+                };
+        }
+    };
+
+    const totalSteps = refreshStepsMockup.steps.length;
+    const numberOfCompletedSteps = refreshStepsMockup.steps.filter((step) => {
+        return step.stepStatus === 'DONE';
+    });
+
+    return (
+        <Drawer
+            autoFocus
+            canEscapeKeyClose
+            canOutsideClickClose
+            enforceFocus
+            hasBackdrop
+            isCloseButtonShown
+            isOpen={isOpen}
+            onClose={() => closeModal(false)}
+            shouldReturnFocusOnClose
+            size={'400px'}
+            title={
+                <RefreshStepsHeadingWrapper className={Classes.DIALOG_HEADER}>
+                    <Icon
+                        icon={statusInfo(refreshStepsMockup.jobStatus).icon}
+                        size={18}
+                    />
+                    <div>
+                        <RefreshStepsTitle>
+                            {statusInfo(refreshStepsMockup.jobStatus).title}
+                        </RefreshStepsTitle>
+                        <StepsCompletionOverview>{`${numberOfCompletedSteps.length}/${totalSteps} steps complete - 00:01:30`}</StepsCompletionOverview>
+                    </div>
+                </RefreshStepsHeadingWrapper>
+            }
+            position={Position.RIGHT}
+        >
+            <StepsWrapper>
+                {refreshStepsMockup.steps.map((step) => (
+                    <Step status={step.stepStatus}>
+                        <div>
+                            {step.stepStatus !== 'PENDING' && (
+                                <StepIcon
+                                    icon={statusInfo(step.stepStatus).icon}
+                                    status={step.stepStatus}
+                                />
+                            )}
```

Comment:
Idea: I think we could avoid the wrapping dic and logic if we set the icon 'none' for the pending status.